test/Translate/Parser/packet_in.p4

```diff
@@ -0,0 +1,72 @@
+// RUN: p4mlir-translate --typeinference-only %s | FileCheck %s
+
+extern packet_in {
+    void extract<T>(out T hdr);
+    void extract<T>(out T variableSizeHeader,
+                    in bit<32> variableFieldSizeInBits);
+    T lookahead<T>();
+    void advance(in bit<32> sizeInBits);
+    bit<32> length();
+}
```

Comment:
Not a request to change this PR.

Now I'm confused whether we can tell a user-defined `packet_in` extern like this, from a `packet_in` extern from `core.p4`? I guess we can't?

Then I wonder how could we treat `packet_in` from `core.p4` differently as discussed in #93?

Reply:
Unfortunately, we cannot. As `core.p4` is included via preprocessor. So we can try somehow to track things to `p4include` dir, but this is pretty error prone. I would probably suggest us to invent some kind of annotation (e.g. `@system` or `@core`) to mark this in the future.

Reply:
Completely agree!

Reply:
The inability to distiniguish system entities from user-defined ones is a long-standing problem in the compiler. Currently, we have the hacky solution where we check the include path, but that can be ambiguous based on user-configuration. 

I am thinking an explicit annotation might be the only way out... 

Maybe a particular special construct defined in the spec can work, too.

Reply:
> The inability to distiniguish system entities from user-defined ones is a long-standing problem in the compiler. Currently, we have the hacky solution where we check the include path, but that can be ambiguous based on user-configuration.

Even more, the behavior might be different with preprocessed sources (!)